### DISCUSSION TITLE
VerneMQ Load Shedding

### DISCUSSION THREAD

**Discussion by tferfce** [vernemq/vernemq#2229] [Q&A] [answered]
Hello,

I have a question regrading VerneMQ configuration. Per documentation [here](https://docs.vernemq.com/configuring-vernemq/options#load-shedding). The online one is set to -1 in our case which means that we do not have any limits on the incoming messages and the offline messages is set to 20000 which limits the number of QOS1 and 2 to 20000 messages. In our case this means that if the MQTT connector remains offline for an amount of time enough to accumulate more than 20K messages in the queue, when the connector reconnects MQTT will drop the excess number of messages resulting in a loss of data. As such we need to understand if `max_offline_messages` is per client, per queue or is general queue size because it is not clearly mentioned in the documentation. 

Thanks,
Thomas

**Comment by mths1** [comment, marked as answer]
Hello,

It is a per queue setting (that can be overwritten in auth_on_register, if needed, iirc). Please also note that the message drop does not happen when the client reconnects, but already when adding to the message queue.

**Comment by ioolkos** [comment]
The `override_max_online_messages = off|on` setting might be of interest, as context.

Usually `max_online_messages` and `max_offline_messages` are configured to the same value, or with offline messages smaller than online. The reason is that the queue would apply the respective limits when changing from offline to online state, potentially dropping some offline messages due to the lower online limit.

`override_max_online_messages` overrides this behaviour and allows all offline messages in the online queue initially. This feature was contributed by @mths1.
@tferfce just for added context, this is probably not an answer to what you are describing.

**Comment by tferfce** [comment]
Thank you both guys! Very much appreciate the help. @ioolkos It is very much related cause its on our options of enabling! 👍
